storybookjs/storybook
# fix: multiple mapped args return array of labels

Issue: https://github.com/storybookjs/storybook/issues/22042

## What I did

Fix issue of configured `mapping` + `control.type=multi-select` argType component is rendered with wrong mapped arg value.

## How to test

- Start a sandbox template: yarn task --task dev --template react-vite/default-ts
- In `sandbox/react-vite-default-ts/src/stories/Button.stories.ts`

```tsx
const arrows = {
  ArrowUp: { name: 'ArrowUp' },
  ArrowDown: { name: 'ArrowDown' },
  ArrowLeft: { name: 'ArrowLeft' },
  ArrowRight: { name: 'ArrowRight' },
};

const meta = {
  title: 'Example/Button',
  component: Button,
  tags: ['autodocs'],
  argTypes: {
    backgroundColor: { control: 'color' },
    arrows: {
      options: Object.keys(arrows),
      mapping: arrows,
      control: {
        type: 'multi-select',
        labels: {
          ArrowUp: 'Up',
          ArrowDown: 'Down',
          ArrowLeft: 'Left',
          ArrowRight: 'Right',
        },
      },
    },
  },
} satisfies Meta<typeof Button>;
```

- Update `Button.tsx` with adding new prop `arrows` and `console.log` it inside `Button` render function
- Verify logged prop `arrows` value is corrected mapped array object
